ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschematools.py

```diff
@@ -0,0 +1,143 @@
+        conversion_mode: str = None,
+        validation_mode: str = None,
+    ) -> None:
+        try:
+            self.avdschema = AvdSchema(schema)
+        except AristaAvdError as e:
+            raise AnsibleActionFail("Invalid Schema!") from e
+
+        self.hostname = hostname
+        self.ansible_display = ansible_display
+        self._set_conversion_mode(conversion_mode)
+        self._set_validation_mode(validation_mode)
+
+    def _set_conversion_mode(self, conversion_mode):
+        if conversion_mode is None:
+            conversion_mode = DEFAULT_CONVERSION_MODE
+
+        if not isinstance(conversion_mode, str):
+            raise AnsibleActionFail("The argument 'conversion_mode' must be a string")
+
+        if conversion_mode not in VALID_CONVERSION_MODES:
+            raise AnsibleActionFail(f"Invalid value '{conversion_mode}' for the argument 'conversion_mode'. Must be one of {VALID_CONVERSION_MODES}")
+
+        self.conversion_mode = conversion_mode
+
+    def _set_validation_mode(self, validation_mode):
+        if validation_mode is None:
+            validation_mode = DEFAULT_VALIDATION_MODE
+
+        if not isinstance(validation_mode, str):
+            raise AnsibleActionFail("The argument 'validation_mode' must be a string")
+
+        if validation_mode not in VALID_VALIDATION_MODES:
+            raise AnsibleActionFail(f"Invalid value '{validation_mode}' for the argument 'validation_mode'. Must be one of {VALID_VALIDATION_MODES}")
+
+        self.validation_mode = validation_mode
+
+    def convert_data(self, data) -> list[str]:
+        """
+        Convert data according to the schema (convert_types)
+        The data conversion is done in-place (updating the original "data" dict).
+
+        avd_schema.convert returns a generator, which we iterate through in handle_exceptions to perform the actual conversions.
+        """
+        if self.conversion_mode == "disabled":
+            return []
+
+        result_messages = []
+        exceptions = self.avdschema.convert(data)
+        if conversion_counter := self.handle_validation_exceptions(exceptions, self.conversion_mode):
+            result_messages.append(f"{conversion_counter} data conversions done to conform to schema.")
+
+        return result_messages
+
+    def validate_data(self, data):
+        """
+        Validate data according to the schema
+
+        avd_schema.validate returns a generator, which we iterate through in handle_exceptions to perform the actual conversions.
+        """
+        if self.validation_mode == "disabled":
+            return []
+
+        result_messages = []
+        exceptions = self.avdschema.validate(data)
+        if validation_counter := self.handle_validation_exceptions(exceptions, self.validation_mode):
+            result_messages.append(f"{validation_counter} errors found during schema validation of input vars.")
+
+        return result_messages
+
+    def convert_and_validate_data(self, data):
```

Comment:
```suggestion
    def convert_and_validate_data(self, data) -> dict:
```

Reply:
Improved type hints in https://github.com/aristanetworks/ansible-avd/pull/2317/commits/72cc53e168325200fe4b0dafee3c66d21206608f